src/TestLogger/Utilities/StringExtensions.cs

```diff
@@ -7,24 +7,6 @@ namespace Spekt.TestLogger.Utilities
 
      public static class StringExtensions
      {
-         public static string ReplaceInvalidXmlChar(this string str)
```

Comment:
So, I completely missed that this was already here, and being used by the Xunit logger. I think this approach is a bit more streamlined overall, even tough its a fairly big change overall. I could fall back on just using this in the junit logger, which would be faster.

Assuming we are going to do keep going this direction, I will go ahead and open PRs in the loggers that depend directly on this branch (temporarily), so we can make sure the API changes aren't breaking anything important. There is a bit of variation across the loggers in what data they accessed, and it looked tricky to keep straight otherwise.